libcloud/common/base.py

```diff
@@ -278,11 +278,14 @@ def response(self):
         if not self._response:
             response = self.connection.connection.getresponse()
             self._response = HttpLibResponseProxy(response)
-            self.body = response.content
             if not self.success():
                 self.parse_error()
         return self._response
 
+    @property
+    def body(self):
+        return self.response.body
```

Comment:
I'm a bit confused, still looking at the test case.

Shouldn't this be `self.response.content` so it results in the same behavior as in the past? Either that, or `self._response.body`, right?

Reply:
I should have explained this since November, 1 2017 and should not have waited for a nice test case, sorry.

Note that `self.response` and `self._response` should be roughly equivalent, except that `self.response` is going to populate `self._response` if it does not exist yet and then return that, so that sounds more robust.

Anyway, `self.response` returns a `HttpLibResponseProxy`, and since I'm trying to follow the law of Demeter I've added a `body` property to `HttpLibResponseProxy` (see below in http.py) to make sure that `RawResponse` does not know about the innards of `HttpLibResponseProxy`. Withouth that property, I would have used `self.response._response.content`. (Note that the variable `response` in the `RawResponse.response` property is a `HttpLibResponseProxy`.)

So that's quite confusing and quite subtle, and this explains why this has been missed in the past!

Hope that helps